docs/source/cc_launcher.md

```diff
@@ -0,0 +1,301 @@
+
+- `bin/detect`: Determine whether or not this buildpack should be used to
+  build the chaincode package and launch it.
+- `bin/build`: Transform the chaincode package into executable chaincode.
+- `bin/release`: Provide metadata to the peer about the chaincode. (optional)
+- `bin/run`: Run the chaincode.  (optional)
+
+#### `bin/detect`
+
+The `bin/detect` script is responsible for determining whether or not a buildpack
+should be used to build a chaincode package and launch it. The peer invokes
+`detect` with two arguments:
+
+```sh
+bin/detect CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR
+```
+
+When `detect` is invoked, `CHAINCODE_SOURCE_DIR` contains the chaincode source
+and `CHAINCODE_METADATA_DIR` contains the `metadata.json` file from the
+chaincode package installed to the peer. If the buildpack should be applied to
+the chaincode source package, `detect` must return an exit code of `0`; any
+other exit code will indicate that the buildpack should not be applied.
+
+The following is an example of a simple `detect` script for go chaincode:
+```sh
+#!/bin/bash
+
+CHAINCODE_METADATA_DIR="$2"
+
+# use jq to extract the chaincode type from metadata.json and exit with
+# success if the chaincode type is golang
+if [ "$(jq -r .type "$CHAINCODE_METADATA_DIR/metadata.json" | tr '[:upper:]' '[:lower:]')" = "golang" ]; then
+    exit 0
+fi
+
+exit 1
+```
+
+#### `bin/build`
+
+The `bin/build` script is responsible for building, compiling, or transforming
+the contents of a chaincode package into artifacts that can be used by
+`release` and `run`. The peer invokes `build` with three arguments:
+
+```sh
+bin/build CHAINCODE_SOURCE_DIR CHAINCODE_METADATA_DIR BUILD_OUTPUT_DIR
+```
+
+When `build` is invoked, `CHAINCODE_SOURCE_DIR` contains the chaincode source
+and `CHAINCODE_METADATA_DIR` contains the `metadata.json` file from the
+chaincode package installed to the peer. `BUILD_OUTPUT_DIR` is the directory
+where `build` must place artifacts needed by `release` and `run`.
+
+When `build` completes with an exit code of `0`, the contents of
+`BUILD_OUTPUT_DIR` will be copied to the persistent storage maintained by the
+peer; any other exit code will be considered a failure.
+
+The following is an example of a simple `build` script for go chaincode:
+```sh
+#!/bin/bash
+
+CHAINCODE_SOURCE_DIR="$1"
+CHAINCODE_METADATA_DIR="$2"
+BUILD_OUTPUT_DIR="$3"
+
+# extract package path from metadata.json
+GO_PACKAGE_PATH="$(jq -r .path "$CHAINCODE_METADATA_DIR/metadata.json")"
+if [ -f "$CHAINCODE_SOURCE_DIR/src/go.mod" ]; then
+    cd "$CHAINCODE_SOURCE_DIR/src"
+    go build -v -mod=readonly -o "$BUILD_OUTPUT_DIR/chaincode" "$GO_PACKAGE_PATH"
+else
+    GO111MODULE=off go build -v  -o "$BUILD_OUTPUT_DIR/chaincode" "$GO_PACKAGE_PATH"
+fi
+
+# save statedb index metadata to provide at release
+if [ -d "$CHAINCODE_SOURCE_DIR/META-INF" ]; then
+    cp -a "$CHAINCODE_SOURCE_DIR/META-INF" "$BUILD_OUTPUT_DIR/"
+fi
+```
+
+#### `bin/release`
+
```

Comment:
We should probably emphasize that if no release binary is present, then this step is skipped.

Reply:
Changed it to `The `bin/release` script is responsible for providing chaincode metadata to the peer. `bin/release` is optional. If it is not provided, this step is skipped.`